tests/build_stitched_index.cpp

```diff
@@ -0,0 +1,855 @@
+                   path &input_data_path, path &final_index_path_prefix,
+                   path &label_data_path, label &universal_label,
+                   unsigned &num_threads, unsigned &R, unsigned &L,
+                   unsigned &stitched_R, float &alpha) {
+  po::options_description desc{"Arguments"};
+  try {
+    desc.add_options()("help,h", "Print information on arguments");
+    desc.add_options()("data_type",
+                       po::value<std::string>(&data_type)->required(),
+                       "data type <int8/uint8/float>");
+    desc.add_options()("data_path",
+                       po::value<path>(&input_data_path)->required(),
+                       "Input data file in bin format");
+    desc.add_options()("index_path_prefix",
+                       po::value<path>(&final_index_path_prefix)->required(),
+                       "Path prefix for saving index file components");
+    desc.add_options()("max_degree,R",
+                       po::value<uint32_t>(&R)->default_value(64),
+                       "Maximum graph degree");
+    desc.add_options()(
+        "Lbuild,L", po::value<uint32_t>(&L)->default_value(100),
+        "Build complexity, higher value results in better graphs");
+    desc.add_options()("stitched_R",
+                       po::value<uint32_t>(&stitched_R)->default_value(100),
+                       "Degree to prune final graph down to");
+    desc.add_options()(
+        "alpha", po::value<float>(&alpha)->default_value(1.2f),
+        "alpha controls density and diameter of graph, set 1 for sparse graph, "
+        "1.2 or 1.4 for denser graphs with lower diameter");
+    desc.add_options()(
+        "num_threads,T",
+        po::value<uint32_t>(&num_threads)->default_value(omp_get_num_procs()),
+        "Number of threads used for building index (defaults to "
+        "omp_get_num_procs())");
+    desc.add_options()("label_file",
+                       po::value<path>(&label_data_path)->default_value(""),
+                       "Input label file in txt format if present");
+    desc.add_options()(
+        "universal_label",
+        po::value<label>(&universal_label)->default_value(UINT32_MAX),
+        "Universal label, if using it, only in conjunction with labels_file");
+
+    po::variables_map vm;
+    po::store(po::parse_command_line(argc, argv, desc), vm);
+    if (vm.count("help")) {
+      std::cout << desc;
+      exit(0);
+    }
+    po::notify(vm);
+  } catch (const std::exception &ex) {
+    std::cerr << ex.what() << '\n';
+    throw;
+  }
+  return 1;
+}
+
+/*
+ * Parses the label datafile, which has comma-separated labels on
+ * each line. Line i corresponds to point id i.
+ *
+ * Returns three objects via std::tuple:
+ * 1. map: key is point id, value is vector of labels said point has
+ * 2. map: key is label, value is number of points with the label
+ * 3. the label universe as a set
+ */
+parse_label_file_return_values parse_label_file(path  label_data_path,
```

Comment:
Isn't there a parsing function in david/filter branch already? Can that be reused?

Reply:
It looks redundant, but the `parse_label_file` in `index.cpp` differs in its treatment of the universal label. For the stitched algorithm, we need to expand the universal label into all labels for bookkeeping purposes; we don't want to do that in the case of membranne, due to performance costs with `std::set_intersection`.